totalRP3/Core/Compression.lua

```diff
@@ -10,36 +9,36 @@ local LibDeflate = LibStub:GetLibrary("LibDeflate");
 
 local Compression = {};
 
-function Compression.compress(data, willBeSentViaAddOnChannel)
-	Ellyb.Assertions.isType(data, "string", "data");
+-- Protocol break for 12.0: Historically, data compression intended for use
+-- in addon messages has been using a codec for regular chat messages. This
+-- incurs a ~53% overhead in transmission size as over half of the range of
+-- bytes needs to be escaped.
+local useAddonChatEncoding = (select(4, GetBuildInfo()) >= 120000);
 
-	local compressedData = LibDeflate:CompressDeflate(data);
+function Compression.compress(data, willBeSentViaAddOnChannel)
+	local compressedData = TRP3_EncodingUtil.CompressString(data);
```

Comment:
I got rid of the assertions in these functions as the native API and LibDeflate already validate types anyway - so we're just duplicating the check needlessly.